spacy/scorer.py

```diff
@@ -873,6 +875,67 @@ def score_deps(
                 f"{attr}_las_per_type": None,
             }
 
+    @staticmethod
+    def score_coref_clusters(examples: Iterable[Example], **cfg):
+        """Score a batch of examples using LEA.
+
+        For details on how LEA works and why to use it see the paper:
+        Which Coreference Evaluation Metric Do You Trust? A Proposal for a Link-based Entity Aware Metric
+        Moosavi and Strube, 2016
+        https://api.semanticscholar.org/CorpusID:17606580
+        """
+
+        span_cluster_prefix = cfg["span_cluster_prefix"]
+
+        evaluator = ClusterEvaluator(lea)
+
+        for ex in examples:
+            p_clusters = doc2clusters(ex.predicted, span_cluster_prefix)
+            g_clusters = doc2clusters(ex.reference, span_cluster_prefix)
+            cluster_info = get_cluster_info(p_clusters, g_clusters)
+            evaluator.update(cluster_info)
+
+        score = {
+            "coref_f": evaluator.get_f1(),
+            "coref_p": evaluator.get_precision(),
+            "coref_r": evaluator.get_recall(),
+        }
+        return score
+
+    @staticmethod
+    def score_span_predictions(examples: Iterable[Example], **cfg):
+        """Evaluate reconstruction of the correct spans from gold heads.
+        """
+        scores = []
+        output_prefix = cfg["output_prefix"]
+        for eg in examples:
+            starts = []
+            ends = []
+            pred_starts = []
+            pred_ends = []
+            ref = eg.reference
+            pred = eg.predicted
+            for key, gold_sg in ref.spans.items():
+                if key.startswith(output_prefix):
+                    pred_sg = pred.spans[key]
+                    for gold_mention, pred_mention in zip(gold_sg, pred_sg):
+                        starts.append(gold_mention.start)
+                        ends.append(gold_mention.end)
+                        pred_starts.append(pred_mention.start)
+                        pred_ends.append(pred_mention.end)
+
+
+            # TODO check logic
+            # see how many are perfect
+            cs = [a == b for a, b in zip(starts, pred_starts)]
+            ce = [a == b for a, b in zip(ends, pred_ends)]
+            correct = [int(a and b) for a, b in zip(cs, ce)]
+            accuracy = sum(correct) / len(correct)
```

Comment:
@kadarakos I had to rewrite this because we can't pass the xp / numpy thing here. I think it's OK but let me know if anything is off about it.

Reply:
In the scorer we only pass the examples and the config, so we don't have access to the `np` associated with the model, and we can't put it in the config because it's not serializable. (Maybe I'm missing something but that's why I assumed this was necessary.) Look at the other functions around this one.